lib/helpers.coffee

```diff
@@ -11,17 +16,88 @@ Helpers = module.exports =
       result.class = result.type.toLowerCase().replace(' ', '-')
       Helpers.validateResults(result.trace) if result.trace
     results
+
+  # Validates that a provider is capable of providing the Base Linter info
+  #   needed to be used at lint time.
   validateLinter: (linter) ->
     unless linter.grammarScopes instanceof Array
-      message = "grammarScopes is not an Array. (see console for more info)"
-      console.warn(message)
-      console.warn('grammarScopes', linter.grammarScopes)
-      throw new Error(message)
-
+      throw new Error("grammarScopes is not an Array. Got: #{linter.grammarScopes})")
     unless linter.lint?
       throw new Error("Missing linter.lint")
-
     if typeof linter.lint isnt 'function'
       throw new Error("linter.lint isn't a function")
-
     return true
+
+  # Everything past this point relates to CLI helpers as loosly demoed out in:
+  #   https://gist.github.com/steelbrain/43d9c38208bf9f2964ab
+
+  exec: (command, options = {stream: 'stdout'}) ->
+    throw new Error "Nothing to execute." if not arguments.length
+    return new Promise (resolve, reject) ->
+      process = child_process.exec(command, options)
+      options.stream = 'stdout' if not options.stream
+      data = []
+      if options.stream == 'stderr'
+        process.stderr.on 'data', (d) -> data.push(d.toString())
+      else
+        process.stdout.on 'data', (d) -> data.push(d.toString())
+      process.stdin.write(options.stdin.toString()) if options.stdin
+      process.on 'close', ->
+        resolve(data)
+
+  # This should only be used if the linter is only working with files in their
+  #   base directory. Else wise they should use `Helpers#exec`.
+  execFilePath: (command, filePath, options = {}) ->
+    throw new Error "Nothing to execute." if not arguments.length
+    throw new Error "No File Path to work with." if not filePath
+    return new Promise (resolve, reject) ->
+      options.cwd = path.dirname(filePath) if not options.cwd
+      command = "#{command} #{filePath}"
+      resolve(Helpers.exec(command, options))
+
+  # Due to what we are attempting to do, the only viable solution right now is
```

Comment:
It's upto you, but IMO having a quick look at the code below automatically describes all of that.